src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    Animal.alive.push(this);
+
+    this.name = name;
+    this.health = 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (!(obj instanceof Carnivore) && obj.hidden === false) {
+      obj.health -= 50;
+
+      if (obj.health === 0) {
+        const diedAnimal = Animal.alive.findIndex(el => el.health === 0);
+
+        Animal.alive.splice(diedAnimal, 1);
```

Comment:
Use `filter` instead of `splice` and `findIndex` and just filter all animals whos health > 0